fastify.js

```diff
@@ -109,12 +110,15 @@ function build (options) {
     server = http.createServer(httpHandler)
   }
 
-  fastify.onClose((instance, done) => {
-    if (listening) {
-      instance.server.close(done)
-    } else {
-      done(null)
-    }
+  app.once('preReady', () => {
```

Comment:
Maybe overwriting avvio ready method fits better here. Or not?

Reply:
Can you rephrase?

Reply:
Putting the `onClose` handler inside `preReady` event doesn't ensure that the callback is invoked before any others. It does only if the `preReady` callbacks are fired in FIFO.

We use

https://github.com/mcollina/avvio/blob/a957201b0b33facf06687a5ba8e52c82a46b1302/boot.js#L13

and creating `ready` method inside fastify.

The new method forces the add `onClose` handler as the last, so the first one to be  invoked.